### PR TITLE
Embed-posting

### PR DESCRIPTION
**Pull request by SummerWish**
The Privly application that supports embed-posting Privly messages.

https://github.com/privly/privly-chrome/pull/120

### CODE REVIEW

**Comment by spectralsun** [comment]
A lot to review here, this may take some time. Good work, though! :smiley:

File: Message/test/new_embed.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  // continuous integration.
+  beforeEach(function () {
+    var keys = Object.keys(__html__);
+    var selectKey;
+    keys.forEach(function (key) {
+      if (key.indexOf("Message/new_embed.html") >= 0) {
+        selectKey = key;
+      }
+    });
+    document.body.innerHTML = __html__[selectKey];
```
Comment (smcgregor) [inline Message/test/new_embed.js]: We are in the process of deprecating the use of the HTML2JS hack. See [Headless Browsers and Fixtures](https://github.com/privly/privly-applications/blob/master/test/README.md)

Comment (smcgregor) [inline Message/test/new_embed.js]: See this pull request in progress https://github.com/privly/privly-applications/pull/238

File: Message/test/new_embed.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+    // Pre-define some mock callback values for Chrome background scripts.
+    // Can be overwritten in test cases.
+    mockValues = {
```
Comment (smcgregor) [inline Message/test/new_embed.js]: global scope?

File: Message/test/new_embed.js (hunk 1 above)
Comment (SummerWish) [inline Message/test/new_embed.js]: Thank you for pointing out this.. thinking about this issue. may need much work. Currently my code logic 'binds' tightly with DOM..

File: Message/test/new_embed.js (hunk 2 above)
Comment (SummerWish) [inline Message/test/new_embed.js]: Oh, it can be changed to test file scope! 😄 I will change it.

**Comment by smcgregor** [comment]
I tested out the functionality and things are mostly working great. :)

One comment: You should only display the anti-spoofing glyph when the user's input will be captured by the extension-context frame. Otherwise the host page will be able to shape the embedded posting/seamless posting iframes into a box that will only show the glyph, then show the glyph wherever they would like to spoof.

I'll proceed with reviewing on the code. Can you please merge in gsoc-develop and open a new pull request against that branch? I'll leave all my comments there.

File: shared/javascripts/options.js [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+    // Set event listeners to execute upgrade() function when
+    // the extension got installed.
+    chrome.runtime.onInstalled.addListener(function () {
+      Privly.options.upgrade();
+    });
+
+    // Listen incoming messages to provide option interfaces
+    // for content scripts
+    chrome.runtime.onMessage.addListener(function (request, sender, sendResponse) {
+      if (request.ask.indexOf('options/') === 0) {
```
Comment (smcgregor) [inline shared/javascripts/options.js]: `request.ask` is not always defined (this broke posting for me).

**Comment by SummerWish** [comment]
Thanks!

This issue is only related to embeded-posting.

In seamless-(embeded)-posting, no glyph will be displayed, instead only the browser modal_button will change (to green icon) when you are inputing in an seamless-posting field. (do you have ideas to improve this? I think a modal_button at the top-right corner is not well-marked)

Heading to merge now ;-)

**Comment by smcgregor** [comment]
I like changing the modal button. We might think about also putting the glyph into the background of the seamless posting form element. Mailvelope does something similar.

**Comment by smcgregor** [comment]
Should I close this and concentrate on #270?

**Comment by SummerWish** [comment]
Yup.

**Pull request closed without merging**